srml/support/src/metadata.rs

```diff
@@ -238,78 +331,83 @@ mod tests {
-							name: DecodeDifferent::Encode("SystemEvent"),
-							arguments: DecodeDifferent::Encode(&[]),
-							documentation: DecodeDifferent::Encode(&[]),
-						}
-					])
+	const EXPECTED_METADATA: RuntimeMetadata = RuntimeMetadata (
+		META_VERSION,
+		RuntimeMetadataV1 {
+		modules: DecodeDifferent::Encode(&[
+			RuntimeModuleMetadata {
+				name: DecodeDifferent::Encode("Module"),
+				prefix: DecodeDifferent::Encode("system"),
+				storage: None,
+				// lost DecodeDifferent::Encode("Module"), aka module metadata name
+				call: DecodeDifferent::Encode(FnEncode(||
+					CallMetadata {
+						name: DecodeDifferent::Encode("Call"),
+						functions: DecodeDifferent::Encode(&[
+						]),
+				})),
+				outer_dispatch: DecodeDifferent::Encode(
+					FnEncode(|| None)
 				),
-				(
-					"event_module",
-					FnEncode(|| &[
-						EventMetadata {
-							name: DecodeDifferent::Encode("TestEvent"),
-							arguments: DecodeDifferent::Encode(&["Balance"]),
-							documentation: DecodeDifferent::Encode(&[" Hi, I am a comment."])
-						}
-					])
+				event: DecodeDifferent::Encode(
+					FnEncode(||
+				 		FnEncode(||&[
+							EventMetadata {
+								name: DecodeDifferent::Encode("SystemEvent"),
+								arguments: DecodeDifferent::Encode(&[]),
+								documentation: DecodeDifferent::Encode(&[])
+							}
+				 		])
+					)
 				),
-				(
-					"event_module2",
-					FnEncode(|| &[
-						EventMetadata {
-							name: DecodeDifferent::Encode("TestEvent"),
-							arguments: DecodeDifferent::Encode(&["Balance"]),
-							documentation: DecodeDifferent::Encode(&[])
-						}
-					])
-				)
-			]),
-		},
-		modules: DecodeDifferent::Encode(&[
+			},
+	
 			RuntimeModuleMetadata {
 				prefix: DecodeDifferent::Encode("event_module"),
-				module: DecodeDifferent::Encode(FnEncode(||
-					ModuleMetadata {
-					 name: DecodeDifferent::Encode("Module"),
-					 call: CallMetadata {
-						 name: DecodeDifferent::Encode("Call"),
-						 functions: DecodeDifferent::Encode(&[
-							 FunctionMetadata {
-								 id: 0,
-								 name: DecodeDifferent::Encode("aux_0"),
-								 arguments: DecodeDifferent::Encode(&[]),
-								 documentation: DecodeDifferent::Encode(&[]),
-							 }
-						 ])
-					 }
-					}
-				)),
+				name: DecodeDifferent::Encode("Module"),
 				storage: None,
+				// lost DecodeDifferent::Encode("Module"), aka module metadata name
+				call: DecodeDifferent::Encode(FnEncode(||
+					CallMetadata {
+						name: DecodeDifferent::Encode("Call"),
+						functions: DecodeDifferent::Encode(&[
+					 		FunctionMetadata {
+						 		id: 0,
+								name: DecodeDifferent::Encode("aux_0"),
+								arguments: DecodeDifferent::Encode(&[]),
+								documentation: DecodeDifferent::Encode(&[]),
+							}
+						]),
+				})),
+				outer_dispatch: DecodeDifferent::Encode(
+					FnEncode(||
+						Some(OuterDispatchCall {
+							name: DecodeDifferent::Encode("EventModule"),
+							prefix: DecodeDifferent::Encode("event_module"),
```

Comment:
We are still repeating stuff, this is always the same as `RuntimeModuleMetadata::prefix`.